src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,211 @@
+
+/**
+ * To create a Sphere Client with retry logic which computes a exponential backoff time delay in milliseconds.
+ * And handle all the configurations for the creation of client.
+ */
+public final class RetryableSphereClientWithExponentialBackoff {
+    protected static final long DEFAULT_MAX_DELAY = 60000;
+    protected static final long DEFAULT_INITIAL_RETRY_DELAY = 200;
+    protected static final int DEFAULT_MAX_RETRY_ATTEMPT = 5;
+    protected static final int DEFAULT_MAX_PARALLEL_REQUESTS = 20;
+    protected static final List<Integer> DEFAULT_STATUS_CODES_TO_RETRY =
+            new ArrayList(Arrays.asList(500, 502, 503, 504));
+
+    private SphereClientConfig clientConfig;
+    private long maxDelay;
+    private long initialRetryDelay;
+    private int maxRetryAttempt;
+    private int maxParallelRequests;
+    private List<Integer> statusCodesToRetry;
+
+    private RetryableSphereClientWithExponentialBackoff(@Nonnull final SphereClientConfig clientConfig) {
+        this.clientConfig = clientConfig;
+        this.maxDelay = DEFAULT_MAX_DELAY;
+        this.initialRetryDelay = DEFAULT_INITIAL_RETRY_DELAY;
+        this.maxRetryAttempt = DEFAULT_MAX_RETRY_ATTEMPT;
+        this.maxParallelRequests = DEFAULT_MAX_PARALLEL_REQUESTS;
+        this.statusCodesToRetry = DEFAULT_STATUS_CODES_TO_RETRY;
+    }
+
+    /**
+     * Creates a new instance of {@link RetryableSphereClientWithExponentialBackoff} given a {@link SphereClientConfig}
+     * responsible for creation of a SphereClient.
+     *
+     * @param clientConfig the client configuration for the client.
+     * @return the instantiated {@link RetryableSphereClientWithExponentialBackoff}.
+     */
+    public static RetryableSphereClientWithExponentialBackoff of(
+            @Nonnull final SphereClientConfig clientConfig) {
+        return new RetryableSphereClientWithExponentialBackoff(clientConfig);
+    }
+
+    /**
+     * Sets the maxDelay value value in milliseconds.
+     * @param maxDelay - build with maxDelay value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given maxDelay value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withMaxDelay(final long maxDelay) {
+        this.maxDelay = maxDelay;
+        return this;
+    }
+
+    /**
+     * Sets the initialDelay value in milliseconds.
+     * @param initialDelay - build with initialDelay value.
+     * If initialDelay is less than maxDelay then, a {@link IllegalArgumentException} will be thrown.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given initialDelay value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withInitialDelay(final long initialDelay) {
+        if (initialDelay < maxDelay) {
+            this.initialRetryDelay = initialDelay;
+        } else {
+            throw new IllegalArgumentException(
+                    format("InitialDelay %s is less than MaxDelay %s.", initialDelay, maxDelay));
```

Comment:
use a standart indentention and reformat all your changes. for intellij ->` pref -> Editor -> Code Style -> Java `

````xml
  <code_scheme name="Default (3)" version="173">
  <JavaCodeStyleSettings>
    <option name="CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND" value="999" />
    <option name="NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND" value="999" />
  </JavaCodeStyleSettings>
  <codeStyleSettings language="JAVA">
    <option name="RIGHT_MARGIN" value="120" />
    <option name="BLANK_LINES_BEFORE_PACKAGE" value="1" />
    <option name="ALIGN_MULTILINE_CHAINED_METHODS" value="true" />
    <option name="SPACE_BEFORE_ARRAY_INITIALIZER_LBRACE" value="true" />
    <option name="SPACE_BEFORE_ANNOTATION_ARRAY_INITIALIZER_LBRACE" value="true" />
    <option name="CALL_PARAMETERS_WRAP" value="1" />
    <option name="METHOD_PARAMETERS_WRAP" value="1" />
    <option name="BINARY_OPERATION_SIGN_ON_NEXT_LINE" value="true" />
    <option name="IF_BRACE_FORCE" value="3" />
    <option name="DOWHILE_BRACE_FORCE" value="3" />
    <option name="WHILE_BRACE_FORCE" value="3" />
    <option name="FOR_BRACE_FORCE" value="3" />
    <indentOptions>
      <option name="CONTINUATION_INDENT_SIZE" value="4" />
    </indentOptions>
  </codeStyleSettings>
</code_scheme>
````

Reply:
Thanks for sharing this code style rules. I have Updated my Intellij.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/623/commits/4b34dc2327c97f31cf4d45daf616edf540dca964